tools/mkfs.c

```diff
@@ -15,6 +15,119 @@
 #include <region.h>
 #include <storage.h>
 
+#define string_ends_with(str, substr) ({                \
+    int slen = strlen(str);                             \
+    int sslen = strlen(substr);                         \
+    boolean res;                                        \
+    if (slen >= sslen)                                  \
+        res = !strcmp((str) + slen - sslen, (substr));  \
+    else                                                \
+        res = false;                                    \
+    res;                                                \
+})
+
+/* Must be large enough to contain the minimum number of clusters (64K) for a
+ * 32-bit File Allocation Table. */
+#define UEFI_PART_SIZE  (33 * MB)
+
```

Comment:
Can we add a comment about where these blobs come from or how they are generated?

Reply:
OK, I added a comment for each blob that says what the blob represents. I also found that I wasn't building the File Allocation Table generically enough, since the number of table entries depends on the file size; this is now fixed in write_uefi_part(), and as a side effect the code is also less cryptic :)